basics/architecture.md

```diff
@@ -163,70 +159,71 @@ For every query, a cluster's broker performs the following:
 }
 ```
 
-**Fault tolerance**
+For multi-stage queries, the broker performs the following:
 
-Broker instances scale horizontally without an upper bound. In a majority of cases, only three brokers are required. If most query results that are returned to a client are less than 1 MB in size per query, you can run a broker and servers inside the same instance container. This lowers the overall footprint of a cluster deployment for use cases that don't need to guarantee a strict SLA on query performance in production.
+* Computes a query plan that runs on multiple sets of servers. The servers selected for the first stage are selected based on the segments required to execute the query, which are determined in a process similar to single-stage queries.
+* Sends the relevant portions of the query plan to each stage of servers in the cluster. A stage may involve one or many servers.
+* The servers that received query plans each execute their part of the query. For more details on this process, read about the [multi-stage engine](https://docs.pinot.apache.org/reference/multi-stage-engine).
+* The broker receives a complete result set from the final stage of the query, which is always a single server.
+* The broker sends the query result to the client.
 
 ### Server
 
-[Servers](components/cluster/server.md) host [segments](components/table/segment/) and do most of the heavy lifting during query processing. Though the architecture shows that there are two kinds of servers, real-time and offline, a server doesn't really "know" if it's going to be a real-time server or an offline server. The server's responsibility depends on the [table](components/table/) assignment strategy.
+[Servers](components/cluster/server.md) host [segments](components/table/segment/) on locally attached storage and perform the computational work of query processing on those segments. By convention, operators speak of "real-time" and "offline" servers, although there is no difference in the server process itself or even its configuration that distinguishes between the two. This is merely a convention reflected in the [table](components/table/) assignment strategy to confine the two different kinds of workloads to two groups of physical instances, since the performance-limiting factors differ between the two kinds of workloads. For example, offline servers might optimize for larger storage capacity, whereas real-time servers might optimize for memory and CPU cores.
```

Comment:
```suggestion
[Servers](components/cluster/server.md) host [segments](components/table/segment/) on locally attached storage and process queries on those segments. By convention, operators speak of "real-time" and "offline" servers, although there is no difference in the server process itself or even its configuration that distinguishes between the two. This is merely a convention reflected in the [table](components/table/) assignment strategy to confine the two different kinds of workloads to two groups of physical instances, since the performance-limiting factors differ between the two kinds of workloads. For example, offline servers might optimize for larger storage capacity, whereas real-time servers might optimize for memory and CPU cores.
```